### PR TITLE
[Internal Review][CL][GaugeDistribute] add functional test distributing coins using CLgauges

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #XXX

## What is the purpose of the change
TestFunctionalConcentratedLiquidityGaugeDistribute is a functional test that covers more complex scenarios relating to distributing incentives throught gauges
at the end of each epoch

we expect these events to occour at the end of each epoch.
- mint and distribute coins according to the configuration
- distribution happens proportional to the gauge weight in the record
- moduleAccount holds the funds and handles distribution

Testing strategy:
1. Initialize variables.
2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
3. Update the gauge by adding rewards (should happen at end of each epoch).
4. get updated gauges and distribute the rewards that was added.
5. Check that incentive has been correctly created and gauge has been correctly updated.

## Testing and Verifying
added functional test 

## Documentation and Release Note
n/a

Where is the change documented? 
n/a

### CODE REVIEW

File: x/incentives/keeper/distribute_test.go [hunk 1]
```diff
@@ -876,3 +887,109 @@ func (s *KeeperTestSuite) TestDistributeConcentratedLiquidity() {
...
+	s.Require().Equal(expectedCoinsToDistribute, actualModuleCoins)
+
+	// 4. get updated gauges and distribute the rewards that was added
+	for _, gauge := range gauges_original {
+		err := s.App.IncentivesKeeper.MoveUpcomingGaugeToActiveGauge(ctx, gauge)
+		s.Require().NoError(err)
+	}
+
+	gauges_afterEpoch := s.App.IncentivesKeeper.GetActiveGauges(ctx)
+	s.Require().Equal(len(gauges_original), len(gauges_afterEpoch))
```
Comment (mattverse) [inline x/incentives/keeper/distribute_test.go:957]: what would this mean? Are we trying to verify that all the given gauges were supposed to be active regardless of `MoveUpcomingGaugeToActiveGauge`?

File: x/incentives/keeper/distribute_test.go [hunk 2]
```diff
@@ -876,3 +887,109 @@ func (s *KeeperTestSuite) TestDistributeConcentratedLiquidity() {
 		})
 	}
 }
+
+// TestFunctionalConcentratedLiquidityGaugeDistribute is a functional test that covers more complex scenarios relating to distributing incentives through gauges
+// at the end of each epoch.
+//
+// we expect these events to occour at the end of each epoch;
+// - mint and distribute coins according to the configuration.
+// - distribution happens proportional to the gauge weight in the record.
```
Comment (mattverse) [inline x/incentives/keeper/distribute_test.go:896]: is this being done in this test?

**Comment by stackman27** [comment]
thank you @mattverse for the review!

holding off on this until we merge https://github.com/osmosis-labs/osmosis/pull/5459, so that we can functional test internal and external gauge

**Comment by stackman27** [comment]
closing this in favor of https://github.com/osmosis-labs/osmosis/pull/5500

**Pull request closed without merging**